File: src/main/java/com/petssocial/pets2/models/Post.java
package com.petssocial.pets2.models;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import javax.persistence.*;
import java.sql.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@ToString
@JsonIgnoreProperties(ignoreUnknown = true)
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String title;
    private String kind;
    private Date date;
    private Date expirationDate;
    private String photo;
    private String optionalField;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private User user;
    private boolean enabled = true;
    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Pet pet;
    private int price;
    @ManyToOne(fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private District postDistrict;
}

File: client/src/app/models/Post.ts
import {Kind} from './enums/Kind';
import {Pet} from './Pet';
import {User} from './User';
import {District} from './District';
import {City} from './City';

export class Post {
  constructor(public title?: string,
              public kind?: Kind,
              public photo?: string,
              public date?: Date,
              public expirationDate?: Date,
              public optionalField?: string,
              public id?: number,
              public pet?: Pet,
              public user?: User,
              public price?: number,
              public postDistrict?: District,
              public enabled?: boolean
              ) {

  }
}
